libs/api/ebpf_api.cpp

```diff
@@ -831,7 +769,7 @@ ebpf_map_set_pin_path(_In_ struct bpf_map* map, _In_ const char* path)
 }
 
 ebpf_result_t
-ebpf_map_unpin(_In_ struct bpf_map* map, _In_opt_z_ const char* path)
+ebpf_map_unpin(_In_opt_ struct bpf_map* map, _In_opt_z_ const char* path)
```

Comment:
Since line 837 always returns an invalid argument error when map is null, do we really want the compiler to allow it without warnings?  Seems like `_In_` is more appropriate since bugs will be caught at compile time not later in runtime.  Hence I recommend reverting this line.